src/resolve/resolved-dns-packet.c

```diff
@@ -810,6 +811,27 @@ int dns_packet_append_opt(
                         goto fail;
 
                 r = dns_packet_append_blob(p, rfc6975, sizeof(rfc6975), NULL);
+
+        } else if (nsid) {
+
+                if (strlen(nsid) > UINT16_MAX - 4) {
+                        r = -E2BIG;
+                        goto fail;
+                }
+
+                r = dns_packet_append_uint16(p, 4 + strlen(nsid), NULL); /* RDLENGTH */
```

Comment:
Hm, in the above `if (edns0_do && include_rfc6975)` block, it seems 'RDLENGTH' info seems not appended. no?

Reply:
it is. line 808 above. i.e. first the 16bit rdlength is appended as sizeof(rfc6975), and the the contents of rfc6975.

I'll add a comment, making this clear

Reply:
Thank you!